Client/Frontend/Browser/URLBarView.swift

```diff
@@ -936,3 +936,50 @@ extension ToolbarTextField: Themeable {
         highlightColor = theme.highlightColor!
     }
 }
+
+class URLBarMinifiedView: BlurWrapper {
+    
+    let touchTarget: UIButton
+    private let title: UILabel
+    private let border: UIView
+    
+    init() {
+        self.touchTarget = UIButton()
+        self.touchTarget.isAccessibilityElement = false
+        
+        self.title = UILabel()
+        self.title.font = UIConstants.DefaultChromeFont
+        self.title.textAlignment = .Center
+        self.title.lineBreakMode = .ByTruncatingMiddle
+        
+        self.border = UIView()
+        self.border.backgroundColor = UIConstants.BorderColor
+        
+        super.init(view: touchTarget)
+        
+        self.addSubview(self.border)
+        self.addSubview(self.title)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func updateConstraints() {
+        super.updateConstraints()
+        
+        self.title.snp_remakeConstraints() { make in
+            make.center.equalTo(self)
+            make.width.equalTo(self).offset(-UIConstants.DefaultPadding)
+        }
+        
+        self.border.snp_remakeConstraints() { make in
+            make.bottom.left.right.equalTo(self)
+            make.height.equalTo(0.5)
+        }
```

Comment:
Nit: Probably don't need to remake the constraints every time the system calls `updateConstraints`. Since the layout doesn't change, you can just setup initial constraints in the constructor and let it ride.